### PR TITLE
Simplify patch handling

### PR DESCRIPTION
**Pull request by agateau-gg**
## Context

While reviewing #905 I thought it was a bit sad that we had to add a field just for backward-compatibility, so I dived into what could be done to get rid of some of the patch special-cases cruft.

## What has been done

Line.content now always contains the full line, including the patch symbol.

This simplifies a bunch of things:
- MatchSpan.from_match() no longer cares about is_patch
- ExtendedMatch.from_match() can reuse match.match as is
- SecretTextOutputHandler no longer needs to reprint the patch symbol
- Line.category is gone, it's replaced by a `is_patch` boolean. This attribute is only used to decide whether to print patch or file line numbers.

If you look at the snapshot diff you will see two things.

First, patch output is more correct: the second `@` at the end of the `@@ ... @@` lines does not miss anymore.

For multiline secrets, the detector left "arm" (the `|___` symbol) starts one character earlier (see the snapshot diffs). This might be fixable but I don't think it's a problem.

## Validation

Tests still pass.

## PR check list

- [x] As much as possible, the changes include tests (unit and/or functional)
- [x] If the changes affect the end user (new feature, behavior change, bug fix) then the PR has a changelog entry (see doc/dev/getting-started.md). If the changes do not affect the end user, then the `skip-changelog` label has been added to the PR.

### CODE REVIEW

File: ggshield/core/lines.py [hunk 1]
```diff
@@ -175,25 +151,22 @@ def get_lines_from_patch(content: str, filemode: Filemode) -> Iterable[Line]:
...
         elif line_type == "-":
             pre_index += 1
             line_pre_index = pre_index
-            line_content = line[1:]
-            category = LineCategory.DELETION
+            line_content = line
         elif line_type == "\\":
             # This type of line shouldn't contain any secret; no need to set indices
-            line_content = line[1:]
+            line_content = line
```
Comment (fnareoh) [inline ggshield/core/lines.py:164]: we can probably simplify the setting of line_content, it is almost always set to line ?

File: ggshield/core/match_span.py [hunk 2]
```diff
@@ -23,39 +23,38 @@ class MatchSpan(NamedTuple):
     column_index_end: int
 
     @staticmethod
-    def from_match(match: Match, lines: List[Line], is_patch: bool) -> "MatchSpan":
+    def from_match(match: Match, lines: List[Line]) -> "MatchSpan":
```
Comment (fnareoh) [inline ggshield/core/match_span.py:26]: :clap: :+1:  it really simplifies this function

**Review by fnareoh — CHANGES_REQUESTED** [review]
Thanks a lot for this, it definitely simplifies the code...
You set this PR to your original fix branch, instead of `garancegourdel/refacto-output-handlers-to-remove-scanable-from-result` is that intentional ? I can take care of the switch if you want/remove my last commit so you can point to it.
You have functional test failing because they capture a new generic secret in the snapshot.

File: ggshield/core/lines.py (hunk 1 above)
Comment (agateau-gg) [reply ggshield/core/lines.py:164]: Good point. Looking into it.

**Pull request merged**